site/content/entry/yuzu-progress-report-feb-2023/index.md

```diff
@@ -0,0 +1,381 @@
+
+Even [your writer](https://github.com/goldenx86) tried to give a hand (emphasis on tried).
+[As reported in the past](https://yuzu-emu.org/entry/yuzu-progress-report-dec-2022/#cpu-requirement-changes-with-free-performance), we applied `Link-time Optimization` to all of yuzu’s subprojects. This meant that every section like audio, input, UI, etc. used LTO at build time, trying to squeeze out every possible bit of performance.
+This caused an unexpected problem: RAM consumption while building increased enough to cause our buildbot to cry for help, causing builds on our Azure DevOps CI to randomly run out of memory.
+
+byte[] suggested that I instead profile which subprojects would provide the most performance boosts, and to only apply LTO to those.
+A few rounds of toasting a CPU by building yuzu later, and it was determined that the two most obvious candidates, core and video_core, were the responsible ones for the performance boost provided by enabling LTO.
+
+{{< gh-hovercard "9872" "Partially applying LTO" >}} to only the core and video_core projects not only reduced compiler RAM use, but also provided a very minor but still consistently measured performance increase, `The Legend of Zelda: Link’s Awakening` improved from 257 FPS to 260 FPS.
+Nothing groundbreaking, but it’s a free bonus from a change that was only intended to reduce RAM use!
+
+[Morph](https://github.com/Morph1984) later added a {{< gh-hovercard "9887" "small but very critical fix" >}} to get all this actually working.
+I’m a disaster, sorry.
+
+## Input improvements
+
+`Metroid Prime Remastered` introduced a serious dilemma: People want to play the game as an FPS with mouse and keyboard.
+
+{{< single-title-imgs
+    "This game is gorgeous, like Samus (Metroid Prime Remastered)"
+    "./mpr.png"
+    "./mpr2.png"
+    >}}
+
+The problem is that the game uses motion on top of stick input to aim, making it an interesting case for how to handle non-controller input.
+
+This was incentive enough to get german77 interested in giving it a try:
+
+- First off, he fixed an issue where {{< gh-hovercard "9757" "motion would be constantly getting recalibrated." >}}
+- Next, german77 tweaked the {{< gh-hovercard "9815" "mouse scaling" >}} for users running a DPI scaling higher than 100% (1.0x).
+- Following up, the {{< gh-hovercard "9842" "mapping for mouse input" >}} was improved. By default a mouse to stick is considered a joystick, which includes assuming deadzone and ranges that aren’t zero by default. This pull request includes some other miscellaneous changes, like fixing some UI elements not working.
+- And finally, to address Metroid Prime Remastered’s `hybrid mode`, support for {{< gh-hovercard "9848" "dedicated motion using the mouse" >}} was added. Since the game only uses two axes, this can be directly mapped 1:1 to mouse movement, giving great ease of aiming!
```

Comment:
```suggestion
- And finally, to address `Metroid Prime Remastered`’s "hybrid mode", support for {{< gh-hovercard "9848" "dedicated motion using the mouse" >}} was added. Since the game only uses two axes, this can be directly mapped 1:1 to mouse movement, making aiming a breeze!
```